src/app/item-page/edit-item-page/item-status/item-status.component.ts

```diff
@@ -92,27 +144,66 @@ export class ItemStatusComponent implements OnInit {
-            return this.authorizationService.isAuthorized(operation.featureID, item.self).pipe(
-              distinctUntilChanged(),
-              map((authorized) => new ItemOperation(operation.operationKey, operation.operationUrl, operation.featureID, !authorized, authorized))
-            );
-          } else {
-            return [operation];
+      // Observable that reads identifiers and their status and, and config properties, and decides
+      // if we're allowed to show a Register DOI feature
+      let showRegister$: Observable<boolean> = combineLatest([this.identifiers$, registerConfigEnabled$]).pipe(
+        distinctUntilChanged(),
+        map(([identifiers, enabled]) => {
+          let no_doi = true;
+          let pending = false;
+          if (identifiers !== undefined && identifiers !== null) {
+            identifiers.forEach((identifier: Identifier) => {
+              if (hasValue(identifier) && identifier.identifierType === 'doi') {
+                // The item has some kind of DOI
+                no_doi = false;
+                if (identifier.identifierStatus === 'PENDING' || identifier.identifierStatus === 'MINTED'
+                  || identifier.identifierStatus == null) {
+                  // The item's DOI is pending, minted or null.
+                  // It isn't registered, reserved, queued for registration or reservation or update, deleted
+                  // or queued for deletion.
+                  pending = true;
+                }
+              }
+            });
           }
-        }),
-        toArray()
-      ).subscribe((ops) => this.operations$.next(ops));
+          // If there is no DOI, or a pending/minted/null DOI, and the config is enabled, return true
+          return ((pending || no_doi) && enabled);
+        })
+      );
+
+      // Subscribe to changes from the showRegister check and rebuild operations list accordingly
+      this.subs.push(showRegister$.subscribe((show) => {
+        // Copy the static array first so we don't keep appending to it
+        let tmp_operations = [...operations];
+        if (show) {
+          // Push the new Register DOI item operation
+          tmp_operations.push(new ItemOperation('registerDOI', this.getCurrentUrl(item) + '/registerdoi', FeatureID.CanRegisterDOI));
+        }
+        // Check authorisations and merge into new operations list
+        observableFrom(tmp_operations).pipe(
+          mergeMap((operation) => {
+            if (hasValue(operation.featureID)) {
+              return this.authorizationService.isAuthorized(operation.featureID, item.self).pipe(
+                distinctUntilChanged(),
+                map((authorized) => new ItemOperation(operation.operationKey, operation.operationUrl, operation.featureID, !authorized, authorized))
+              );
+            } else {
+              return [operation];
+            }
+          }),
+          toArray()
+        ).subscribe((ops) => this.operations$.next(ops));
```

Comment:
Probably we should refactor this part of code avoiding to have nested subscription

Reply:
@atarix83 can you please give use more details about how to refactor this? We thought this is a good way to do this.

Reply:
i believe this also follows the same pattern as the version currently in main, but if there are optimisation or nesting issues that we need to address, happy to look at it - I might just need some further info or pointers from the angular dev perspective.

As per main https://github.com/DSpace/dspace-angular/blob/main/src/app/item-page/edit-item-page/item-status/item-status.component.ts:
```js
      observableFrom(operations).pipe(
        mergeMap((operation) => {
          if (hasValue(operation.featureID)) {
            return this.authorizationService.isAuthorized(operation.featureID, item.self).pipe(
              distinctUntilChanged(),
              map((authorized) => new ItemOperation(operation.operationKey, operation.operationUrl, operation.featureID, !authorized, authorized))
            );
          } else {
            return [operation];
          }
        }),
        toArray()
      ).subscribe((ops) => this.operations$.next(ops));
```

Reply:
@kshepherd : To provide a hint here... a "nested subscription" is when you have a "subscribe" inside another "subscribe".  This subscribe (line 197) is nested under another subscribe (line 177).  The example code you posted in your comment does *not* appear to have this same issue... it's an example of the better way to code this.

I'll admit, I'm a newbie myself when it comes to RxJS, but there are docs out there about the dangers of nested subscriptions.  Here's a few:
https://blog.brecht.io/rxjs-best-practices-in-angular/#avoiding-nested-subscribes
https://vocon-it.com/2018/09/04/how-to-get-rid-of-nested-subscriptions-using-rxjs-flatmap-and-switchmap/

I'm not the right person to provide advice on how exactly to refactor this yet (I'm still learning too), but I'm hoping this will provide you with the hints you need to work it out.

Reply:
thanks a lot @artlowel ! I'll give it a try today and let you know how i go.

Reply:
@artlowel The code compiles OK, and I can see the expected things happening in the map callbacks, but the subscribe never catches anything emitted... it seems that `toArray()` is never reached, which could be some issue where the source observable is never properly completed, from what I can see... I will keep digging and look into how I can debug the whole subscription pipeline

Reply:
`toArray()` only emits if all the observables before have completed, so perhaps you need a `take(1)` at the end of the `this.authorizationService.isAuthorized` pipe to ensure it completes? That would be surprising though as the original code used `toArray()` too, but it's worth a shot.

Something to help debug could be to add a `tap((v) => console.log('something to identify the log', v)),` at any place in the pipe you want log what's passing by at that point.

Reply:
@artlowel thanks. After some debugging with tap, and different pipe structures, I settled on something which I think does the job well and simplifies some of the logic, as well as avoiding nested subscriptions. I think part of the problem with the toArray / emit completion was the way I was reading the REST configuration property.